overlord/snapstate/snapmgr.go

```diff
@@ -633,6 +634,37 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// GenerateCookies creates snap cookies for snaps that are missing them (may be the case for snaps installed
+// before the feature of running snapctl outside of hooks was introduced, leading to a warning
+// from snap-confine).
+func (m *SnapManager) GenerateCookies(st *state.State) error {
+	var snapNames map[string]*json.RawMessage
+	err := st.Get("snaps", &snapNames)
+	if err != nil && err != state.ErrNoState {
```

Comment:
This could be written in a single line: `if err := ...; err != nil && err != state.ErrNoState {`